tests/unit/alp_cholesky.cpp

```diff
@@ -22,71 +22,169 @@
 #include <alp.hpp>
 #include <alp/algorithms/cholesky.hpp>
 #include <alp/utils/parser/MatrixFileReader.hpp>
+#include "../utils/print_alp_containers.hpp"
 
 using namespace alp;
 
-void alp_program( const size_t & unit, alp::RC & rc ) {
-	rc = SUCCESS;
+typedef double ScalarType;
+constexpr ScalarType tol = 1.e-12;
+
+struct inpdata {
+	std::string fname;
+};
+
+//** check the solution by calculating the Frobenius norm of (H-LL^T) **//
+template< typename T, typename RingType, typename ZeroType >
+alp::RC check_cholesky_solution(
+	const  alp::Matrix< T, structures::Symmetric, Dense > &H,
+	alp::Matrix< T, structures::UpperTriangular, Dense > &L,
+	const ZeroType &zero_scalar,
+	const RingType &ring
+) {
+	alp::RC rc = SUCCESS;
+	const size_t N = nrows( H );
+	alp::Matrix< T, alp::structures::Symmetric > LLT( N, N );
+	alp::set( LLT, zero_scalar );
```

Comment:
Should we assign the return value of `set` to `rc` to detect a potential failure?